### DISCUSSION TITLE
Seperate searX from SearXNG

### DISCUSSION THREAD

**Discussion by ManeraKai** [searxng/searxng#1046] [Q&A] [answered]
This move will help us a lot in [LibRedirect](https://github.com/libredirect/libredirect).
Context: https://github.com/libredirect/libredirect/issues/120

**Comment by return42** [comment]
Are you asking for separate the namespace of SearXNG from `searx` ..

- https://github.com/searxng/searxng/pull/358
- https://github.com/searxng/searxng/issues/425

.. or do you ask for distinguish searx & SearXNG instances on https://searx.space/

- https://github.com/searxng/searx-space/issues/104

**Comment by ManeraKai** [comment]
To distinguish.

> https://github.com/searxng/searx-space/issues/104

oh i didn't know he made an issue.

**Comment by dalf** [comment, marked as answer]
See https://github.com/searxng/searx-space/pull/105 : each instance is going to contain a "generator" field.

**Reply by ManeraKai** [reply]
Interesting. So it will be included in a `<meta>` element that it's either a searx or a searxng right?

**Reply by tiekoetter** [reply]
@ManeraKai The `<meta name="generator">` tag is read from each instance and included as a `generator` key in instances.json

**Reply by ManeraKai** [reply]
It's not in the [instances.json](https://searx.space/data/instances.json)

**Reply by tiekoetter** [reply]
It needs to update first.

**Reply by dalf** [reply]
Once searx.space is updated with this PR (in few hours), you will be able to update the code of libredirect with something like this:
https://github.com/libredirect/libredirect/blob/00675d7772476be2cc2c07a3208c600a48489ca9/src/instances/get_instances.py#L151 : 

```python
searxList = {}
for item in rJson['instances'].keys():
    searxInstanceSubList = searxList.setdefault(item.get('generator', 'searx'))
    item = item[:-1]
    network = 'normal'
    if item.endswith('.onion'):
        network = 'tor'
    elif item.endswith('.i2p'):
        network = 'i2p'
    searxInstanceSubList.setdefault(network).append(item)
```
-->
* `searxList["searx"]["normal"]`, `searxList["searx"]["tor"]`, `searxList["searx"]["i2p"]`
* `searxList["searxng"]["normal"]`, `searxList["searxng"]["tor"]`, `searxList["searxng"]["i2p"]`

**Reply by dalf** [reply]
It's on https://searx.space/data/instances.json
